src/Service.Tests/SqlTests/RestApiTests/Patch/MsSqlPatchApiTests.cs

```diff
@@ -190,6 +216,101 @@ public async Task PatchOneViewBadRequestTest()
             await base.PatchOneViewBadRequestTest(expectedErrorMessage);
         }
 
+        /// <summary>
+        /// Test to validate successful execution of a request when a rowversion field is missing from the request body.
+        /// </summary>
+        [TestMethod]
+        public async Task PatchOneWithRowversionFieldMissingFromRequestBody()
+        {
+            // Validate successful execution of a PATCH update when a rowversion field (here 'row_version')
+            // is missing from the request body. A PATCH update request should not try to update any field which is not provided in the request body.
+            string requestBody = @"
+            {
+                ""book_name"": ""Another Awesome Book"",
+                ""copies_sold"": 100,
+                ""last_sold_on"": null
+            }";
+            string expectedLocationHeader = $"id/1";
+
+            await SetupAndRunRestApiTest(
+                    primaryKeyRoute: expectedLocationHeader,
+                    queryString: null,
+                    entityNameOrPath: _entityWithReadOnlyFields,
+                    sqlQuery: GetQuery("PatchOneUpdateWithRowversionFieldMissingFromRequestBody"),
+                    operationType: EntityActionOperation.UpsertIncremental,
+                    requestBody: requestBody,
+                    expectedStatusCode: HttpStatusCode.OK
+                );
+
+            // Validate successful execution of a PATCH insert when a rowversion field (here 'row_version')
+            // is missing from the request body.Successful execution of the PATCH request confirms that we did not
+            // attempt to NULL out the 'row_version' field while inserting the record.
+            requestBody = @"
+            {
+                ""book_name"": ""Best seller"",
+                ""copies_sold"": 100,
+                ""last_sold_on"": null
+            }";
+
+            expectedLocationHeader = $"id/2";
+
+            await SetupAndRunRestApiTest(
+                    primaryKeyRoute: expectedLocationHeader,
+                    queryString: null,
+                    entityNameOrPath: _entityWithReadOnlyFields,
+                    sqlQuery: GetQuery("PatchOneInsertWithRowversionFieldMissingFromRequestBody"),
```

Comment:
`PatchOneInsertWithRowversionFieldMissingFromRequestBody` doesn't check the row_version field.
The comment for this particular test is still unclear about how you're validating that the patch request did not attempt to null out `row_version` field. Stating "successful execution of X confirms" doesn't really capture what I'm asking. You'll still need to answer **why** succesful execution indicates DAB didn't attempt to null out the field. Let's say someone updates the code to actually modify the value/attempt to modify the value: how would this test fail?

Reply:
https://github.com/Azure/data-api-builder/pull/1596#discussion_r1322397055